File: 1768_Merge_Strings_Alternately.go
package main

import(
    "fmt"
    "strings"
)

func mergeAlternately(word1 string, word2 string) string {
    var i,j int //Track position in each word
    var char,fout string

    if word1 == ""{
        return word2
    } else if word2 == "" {
        return word1
    } else {
        sword1 := strings.Split(word1,"") //Convert string into array
        sword2 := strings.Split(word2,"")

        for t := 0; t < len(word1)+len(word2); t++ {
            if t%2 == 0{
                char = sword1[i]
                fout = fmt.Sprintf("%s%s",fout,char)
                i++
                if i >= len(word1){
                    fout = fmt.Sprintf("%s%s",fout,word2[j:]) //Rest of second word
                    break
                }
            } else {
                char = sword2[j]
                fout = fmt.Sprintf("%s%s",fout,char)
                j++
                if j >= len(word2){
                    fout = fmt.Sprintf("%s%s",fout,word1[i:])//Rest of first word
                    break
                }
            }
        }
    return fout
    }
}

File: 1768_Merge_Strings_Alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """

        #Initialization
        fout = ""
        t = 0 #Ticker to alternate between words
        i = 0 #Index on first word
        j = 0 #Index on second word
        foutlength = len(word1 + word2) #Brute appending to check output length            

        while t < foutlength:
            if t%2 == 0:
                fout = fout + word1[i]
                i += 1
                if i > len(word1)-1:
                    return fout + word2[j:]
            else:
                fout = fout + word2[j]
                j += 1
                if j > len(word2)-1:
                    return fout + word1[i:]
            t = t+1
            
